napari/plugins/io.py

```diff
@@ -60,6 +60,14 @@ def read_data_with_plugins(
     """
     hook_caller = plugin_manager.hook.napari_get_reader
 
+    # this checks for the "null layer" sentinel [(None, )]
+    def is_null_layer_sentinel(_layer_data):
```

Comment:
probably overcautious, but should these be `isinstance(_layer_data, list)` and `tuple`?  `len(_layer_data)` will raise an unrelated `TypeError` if it encounters a non-iterable

Reply:
also, let's move this function out of the body of `read_data_with_plugins` while we're at it

Reply:
good catch